elements/ip/flowrrswitch.hh

```diff
@@ -0,0 +1,159 @@
+#ifndef CLICK_FLOWRRSWITCH_HH
+#define CLICK_FLOWRRSWITCH_HH
+
+#include <click/batchelement.hh>
+#include <click/ipflowid.hh>
+#include <click/hashtable.hh>
+
+CLICK_DECLS
+
+/*
+ * =c
+ * FlowRRSwitch()
+ *
+ * =s classification
+ * splits flows across its ports in a round-robin fashion
+ *
+ * =d
+ * Can have any number of outputs.
+ * Chooses the output on which to emit each flow based on
+ * a round-robin scheme across the number of output ports.
+ * The element use a hashtable to remember each 5-tuple
+ * passing by and is therefore *not* thread-safe.
+ *
+ * =e
+ * This element expects IP packets and chooses the output
+ * by applying round-robin
+ *
+ *   FlowRRSwitch()
+ * =a
+ * Switch, HashSwitch, StrideSwitch, RandomSwitch,
+ */
+class FlowRRSwitch : public BatchElement {
+
+    public:
+
+        FlowRRSwitch() CLICK_COLD;
+        ~FlowRRSwitch() CLICK_COLD;
+
+        const char *class_name() const { return "FlowRRSwitch"; }
+        const char *port_count() const { return "1/1-"; }
+        const char *processing() const { return PUSH; }
+
+        int configure(Vector<String> &, ErrorHandler *) CLICK_COLD;
+        void cleanup(CleanupStage) CLICK_COLD;
+        void add_handlers() CLICK_COLD;
+
+        static String read_handler(Element *handler, void *user_data);
+
+        int process(int port, Packet *);
+        void push(int port, Packet *);
+    #if HAVE_BATCH
+        void push_batch(int port, PacketBatch *);
+    #endif
+
+    private:
+
+        class IPFlow {
```

Comment:
This same class is in GenerateIPPacketRules.
Can we somehow decouple it and reuse it across multiple elements?

Reply:
That's in Metron. When merging we can factorize. Just a new file in include/click. But the name should be more descriptive. Why the size mostly? And IPFlowID has already a proto field also. 
Maybe even better : simply use the flow subsystem. So just extend FlowSpaceElement<PortSizeTuple> and need to have a FlowIPManager in the path and done. I'll merge this version and we can change that later.